src/dev/impl/DevToys/ViewModels/Tools/Generators/CheckSumGenerator/CheckSumGeneratorToolViewModel.cs

```diff
@@ -0,0 +1,429 @@
+
+        public IAsyncRelayCommand SelectFilesBrowseCommand { get; }
+
+        private async Task ExecuteSelectFilesBrowseCommandAsync()
+        {
+            var filePicker = new FileOpenPicker
+            {
+                ViewMode = PickerViewMode.List,
+                SuggestedStartLocation = PickerLocationId.ComputerFolder,
+            };
+            filePicker.FileTypeFilter.Add("*");
+
+            await ThreadHelper.RunOnUIThreadAsync(async () =>
+            {
+                StorageFile file = await filePicker.PickSingleFileAsync();
+                if (file != null)
+                {
+                    InputFile = file;
+                }
+            });
+        }
+
+        #endregion
+
+        #region SelectFilesPasteCommand
+
+        public IAsyncRelayCommand SelectFilesPasteCommand { get; }
+
+        private async Task ExecuteSelectFilesPasteCommandAsync()
+        {
+            await ThreadHelper.RunOnUIThreadAsync(async () =>
+            {
+                DataPackageView? dataPackageView = Clipboard.GetContent();
+
+                IStorageItem? storageItem = await ExtractStorageItem(dataPackageView);
+                if (storageItem is StorageFile file)
+                {
+                    InputFile = file;
+                }
+            }).ConfigureAwait(false);
+        }
+
+        #endregion
+
+        #region CancelCommand
+
+        public IRelayCommand CancelCommand { get; }
+
+        private void ExecuteCancelCommand()
+        {
+            lock (_lockObject)
+            {
+                _hasCancelledCalculation = true;
+                _cancellationTokenSource.Cancel();
+                _cancellationTokenSource.Dispose();
+
+                _cancellationTokenSource = new CancellationTokenSource();
+
+                if (_outputHashingAlgorithm != null)
+                {
+                    InputHashingAlgorithm = _outputHashingAlgorithm;
+                }
+
+                _isCalculationInProgress = false;
+                _hasCancelledCalculation = false;
+                ShouldDisplayProgress = false;
+                ResetProgress();
+            }
+        }
+
+        #endregion
+
+        private void UpdateOutputCase()
+        {
+            if (string.IsNullOrWhiteSpace(Output))
+            {
+                return;
+            }
+
+            if (IsUppercase)
+            {
+                Output = Output!.ToUpperInvariant();
+            }
+            else
+            {
+                Output = Output!.ToLowerInvariant();
+            }
+        }
+
+        private void CheckSum()
+        {
+            lock (_lockObject)
+            {
+                ResetProgress();
+                CheckSumAsync().Forget();
+            }
+        }
+
+        private async Task CheckSumAsync()
+        {
+            if (_isCalculationInProgress || InputFile is null)
+            {
+                return;
+            }
+
+            _isCalculationInProgress = true;
+
```

Comment:
Please call `await TaskScheduler.Default;` here. This allows to force the method to go out of the UI thread, which guarantees us to not hanging the UI until we call RunOnUIThreadAsync

Reply:
Good point, added it!